librz/bin/format/mach0/dyldcache.c

```diff
@@ -1,295 +1,1084 @@
+				continue;
+			}
+			deps[dep_index]++;
+			eprintf("-> %s\n", key);
+		}
+		cursor += cmdsize;
+	}
+
+beach:
+	free(cmds);
+}
+
+static RzList *create_cache_bins(RzBuffer *cache_buf, cache_hdr_t *hdr, cache_map_t *maps, cache_accel_t *accel) {
+	RzList *bins = rz_list_newf((RzListFree)free_bin);
+	if (!bins) {
+		return NULL;
+	}
+
+	cache_img_t *img = read_cache_images(cache_buf, hdr);
+	if (!img) {
+		rz_list_free(bins);
+		return NULL;
+	}
+
+	int i;
+	int *deps = NULL;
+	char *target_libs = NULL;
+	target_libs = rz_sys_getenv("RZ_DYLDCACHE_FILTER");
+	RzList *target_lib_names = NULL;
+	ut16 *depArray = NULL;
+	cache_imgxtr_t *extras = NULL;
+	if (target_libs) {
+		target_lib_names = rz_str_split_list(target_libs, ":", 0);
+		if (!target_lib_names) {
+			goto error;
+		}
+
+		deps = RZ_NEWS0(int, hdr->imagesCount);
+		if (!deps) {
+			goto error;
+		}
+
+		HtPU *path_to_idx = NULL;
+		if (accel) {
+			depArray = RZ_NEWS0(ut16, accel->depListCount);
+			if (!depArray) {
+				goto error;
+			}
+
+			if (rz_buf_fread_at(cache_buf, accel->depListOffset, (ut8 *)depArray, "s", accel->depListCount) != accel->depListCount * 2) {
+				goto error;
+			}
+
+			extras = read_cache_imgextra(cache_buf, hdr, accel);
+			if (!extras) {
+				goto error;
+			}
+		} else {
+			path_to_idx = create_path_to_index(cache_buf, img, hdr);
+		}
+
+		for (i = 0; i < hdr->imagesCount; i++) {
+			char *lib_name = get_lib_name(cache_buf, &img[i]);
+			if (!lib_name) {
+				break;
+			}
+			if (strstr(lib_name, "libobjc.A.dylib")) {
+				deps[i]++;
+			}
+			if (!rz_list_find(target_lib_names, lib_name, string_contains)) {
+				RZ_FREE(lib_name);
+				continue;
+			}
+			eprintf("FILTER: %s\n", lib_name);
+			RZ_FREE(lib_name);
+			deps[i]++;
+
+			if (extras && depArray) {
+				ut32 j;
+				for (j = extras[i].dependentsStartArrayIndex; depArray[j] != 0xffff; j++) {
+					ut16 dep_index = depArray[j] & 0x7fff;
+					deps[dep_index]++;
+
+					char *dep_name = get_lib_name(cache_buf, &img[dep_index]);
+					if (!dep_name) {
 						break;
 					}
-					if (sect.offset > libsz) {
-						rz_bin_dyldcache_apply_patch(dbuf, sect.offset - sect_offset,
-							addend + rz_offsetof(struct section, offset));
+					eprintf("-> %s\n", dep_name);
+					free(dep_name);
+				}
+			} else if (path_to_idx) {
+				carve_deps_at_address(cache_buf, img, hdr, maps, path_to_idx, img[i].address, deps);
+			}
+		}
+
+		ht_pu_free(path_to_idx);
+		RZ_FREE(depArray);
+		RZ_FREE(extras);
+		RZ_FREE(target_libs);
+		rz_list_free(target_lib_names);
+		target_lib_names = NULL;
+	}
+
+	for (i = 0; i < hdr->imagesCount; i++) {
+		if (deps && !deps[i]) {
+			continue;
+		}
+		ut64 pa = va2pa(img[i].address, hdr, maps, cache_buf, 0, NULL, NULL);
+		if (pa == UT64_MAX) {
+			continue;
+		}
+		ut8 magicbytes[4];
+		rz_buf_read_at(cache_buf, pa, magicbytes, 4);
+		int magic = rz_read_le32(magicbytes);
+		switch (magic) {
+		case MH_MAGIC:
+			// parse_mach0 (ret, *ptr, bf);
```

Comment:
Is there any reason why this commented?

Reply:
32bit is not supported (yet), I'll make the comment more meaningful.